ampd/src/mvx/mod.rs

```diff
@@ -76,6 +82,10 @@ fn uint256_to_compact_vec(value: Uint256) -> Vec<u8> {
     bytes[slice_from..].to_vec()
 }
 
+fn usize_to_u32(value: usize) -> Vec<u8> {
+    (value as u32).to_be_bytes().to_vec()
+}

```

Comment:
I'd prefer to move the `to_vec()` call out of this function and return a fixed length byte array. This being fixed length is security relevant, so I wouldn't hide it behind a vec conversion

Reply:
Fixed.